src/api/cvc4cpp.h

```diff
@@ -0,0 +1,2410 @@
+     * @return true if the sort is the string sort
+     */
+    bool isString() const;
+
+    /**
+     * Is this a regexp sort?
+     * @return true if the sort is the regexp sort
+     */
+    bool isRegExp() const;
+
+    /**
+     * Is this a rounding mode sort?
+     * @return true if the sort is the rounding mode sort
+     */
+    bool isRoundingMode() const;
+
+    /**
+     * Is this a bit-vector sort?
+     * @return true if the sort is a bit-vector sort
+     */
+    bool isBitVector() const;
+
+    /**
+     * Is this a floating-point sort?
+     * @return true if the sort is a floating-point sort
+     */
+    bool isFloatingPoint() const;
+
+    /**
+     * Is this a datatype sort?
+     * @return true if the sort is a datatype sort
+     */
+    bool isDatatype() const;
+
+    /**
+     * Is this a function sort?
+     * @return true if the sort is a function sort
+     */
+    bool isFunction() const;
+
+    /**
+     * Is this a predicate sort?
+     * That is, is this a function sort mapping to Boolean? All predicate
+     * sorts are also function sorts.
+     * @return true if the sort is a predicate sort
+     */
+    bool isPredicate() const;
+
+    /**
+     * Is this a tuple sort?
+     * @return true if the sort is a tuple sort
+     */
+    bool isTuple() const;
+
+    /**
+     * Is this a record sort?
+     * @return true if the sort is a record sort
+     */
+    bool isRecord() const;
+
+    /**
+     * Is this an array sort?
+     * @return true if the sort is a array sort
+     */
+    bool isArray() const;
+
+    /**
+     * Is this a Set sort?
+     * @return true if the sort is a Set sort
+     */
+    bool isSet() const;
+
+    /**
+     * Is this a sort kind?
+     * @return true if this is a sort kind
+     */
+    bool isUninterpretedSort() const;
+
+    /**
+     * Is this a sort constructor kind?
+     * @return true if this is a sort constructor kind
+     */
+    bool isSortConstructor() const;
+
+    /**
+     * @return the underlying datatype of a datatype sort
+     */
+    Datatype getDatatype() const;
+
+    /**
+     * Instantiate a parameterized datatype/sort sort.
+     * Create sorts parameter with Solver::mkParamSort().
+     * @param params the list of sort parameters to instantiate with
+     */
+    Sort instantiate(const std::vector<Sort>& params) const;
+
+    /**
+     * Output a string representation of this sort to a given stream.
+     * @param out the output stream
+     */
+    void toStream(std::ostream& out) const;
+
+    /**
+     * @return a string representation of this sort
+     */
+    std::string toString() const;
+
+  private:
+    /**
+     * Constructor.
+     * @param t the internal type that is to be wrapped by this sort
+     * @return the Sort
+     */
+    Sort(const CVC4::Type& t);
+
+    /**
+     * The interal type wrapped by this sort.
+     * This is a unique_ptr rather than a shared_ptr since CVC4::Type is
+     * already ref counted.
+     */
```

Comment:
Here and below (Term etc):

Using a unique_ptr means that memory is allocated every time a Sort is copied.  I'm not sure this is a good idea.  shared_ptr would avoid this, right?